lighthouse-core/audits/dobetterweb/charset.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @license Copyright 2016 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/**
+ * @fileoverview Audits a page to ensure charset it configured properly.
+ * It must be defined within the first 1024 bytes of the HTML document, defined in the HTTP header, or in a BOM.
+ */
+'use strict';
+
+const Audit = require('../audit.js');
+const i18n = require('../../lib/i18n/i18n.js');
+const MainResource = require('../../computed/main-resource.js');
+const CONTENT_TYPE_HEADER = 'content-type';
+const CHARSET_META_REGEX = /<meta.*charset="?.{1,}"?.*>/gm;
+const CHARSET_HTTP_REGEX = /charset=.{1,}/gm;
```

Comment:
i mentioned you could add these consts to the module.exports, and that way you can write some unit tests against them.

iirc, you wrote these regexs to handle some extra fancy cases that the current unit tests dont cover.  like `charset=` (empty string value).   so i think it'd be worth having a unit test for each regex just to give it a handful of cases it should match and not match.

Reply:
> that way you can write some unit tests against them.

let me know if you have any questions on this.  i'm thinking 1 test with like 10-ish assertions using various html variants.
regexes are hilarious so its good to test out all sorts of edge cases.